crates/iota-data-ingestion/src/workers/blob.rs

```diff
@@ -41,29 +41,17 @@ where
 }
 
 pub struct BlobWorker {
-    remote_store: Box<dyn ObjectStore>,
+    remote_store: Arc<DynObjectStore>,
     checkpoint_chunk_size_mb: u64,
 }
 
 impl BlobWorker {
-    pub fn new(config: BlobTaskConfig) -> Self {
-        Self {
+    pub fn try_new(config: BlobTaskConfig) -> anyhow::Result<Self> {
```

Comment:
Nit: `new` may very well be fallible.

Reply:
[fixed](https://github.com/iotaledger/iota/pull/5155/commits/fa4c05edab4333eaa6447b5ff5951263d81b6b85)